### DISCUSSION TITLE
Menu / Navbar: How to apply style to selected menu item?

### DISCUSSION THREAD

**Discussion by gitmalong** [leptos-rs/leptos#1559] [Q&A] [answered]
Hi!

Given a navbar with Menu Entries A, B and C I could create three boolean signals to determine if a menu entry is currently selected and setting this var to false for all not selected ones when an item is clicked. I guess this is a pretty ugly way of "marking the currently opened menu item".

Another way would be to check the current URL against a menu item's value. But how do I get the current URL Path in Leptos from my Component?

Thanks

**Comment by gbj** [comment, marked as answer]
This sounds to me like a very good use case for the [`<A/>` component](https://docs.rs/leptos_router/latest/leptos_router/fn.A.html), and if not, then [the logic that `<A/>` uses to check if it's active](https://docs.rs/leptos_router/latest/src/leptos_router/components/link.rs.html#115) may be useful for you to just reuse.
